ultralytics/utils/tlc/utils.py

```diff
@@ -119,7 +120,24 @@ def check_tlc_dataset(
     value_map = tables[first_split].get_value_map(label_column_name)
     names = {int(k): v['internal_name'] for k, v in value_map.items()}
 
-    return {**tables, "names": names, "nc": len(names)}
+    for split in tables:
+        if tables[split].get_value_map(label_column_name) != value_map:
+            msg = f"All splits must have the same categories, but {split} has different categories from {first_split}."
+            raise ValueError(msg)
+
+    # Map name indices to 0, 1, ..., n-1
+    names_yolo = dict(enumerate(names.values()))
+    range_to_3lc_class = dict(enumerate(names))
+
+    return {
+        **tables,
+        "names": names_yolo,
+        "names_3lc": names,
+        "nc": len(names),
+        "range_to_3lc_class": range_to_3lc_class,
```

Comment:
@gudbrandtandberg do you have an idea for a better name for this?

Reply:
I think it is fine